UI

JirkaVrbka opened the pull request.

JirkaVrbka committed:
Removed unused imports

JirkaVrbka committed:
Added getAll

JirkaVrbka committed:
Vue installation

JirkaVrbka committed:
added api prefix

JirkaVrbka committed:
Index page

JirkaVrbka committed:
styling

JirkaVrbka committed:
Fixed user registration

JirkaVrbka committed:
Updated api

JirkaVrbka committed:
created basic UI

JirkaVrbka committed:
Removed unused script

JirkaVrbka committed:
Added UI execution

JirkaVrbka committed:
transparency

JirkaVrbka committed:
Dependencies

JirkaVrbka committed:
Navbar to navbar

Michal-MK commented on README.md:
```diff
@@ -16,6 +16,23 @@ The service keeps records of customers and their:
+ To run UI you have to:
+  * Open terminal
+  * Go to 'ui\src\frontend'
+  * execute command 'npm run serve'
```
I had to run `npm install` before the `serve` command, then it worked

Michal-MK commented on api/src/main/java/cz/idomatojde/facade/OfferFacade.java:
```diff
@@ -5,6 +5,7 @@
 import cz.idomatojde.dto.offer.OfferDTO;
```
Accidental change? Nothing was added except unused import and a blank line.

Michal-MK commented on business/src/main/java/cz/idomatojde/services/facades/OfferFacadeImpl.java:
```diff
@@ -60,4 +60,6 @@ public void changeDescription(ChangeDescriptionOfferDTO changeDescriptionOfferDT
         List<Offer> offers = offerService.getOffersByCategory(mapService.mapDto(category, Category.class));
         return mapService.mapEntityCollection(new ArrayList<>(offers), OfferDTO.class);
     }
+
```
Blank lines

Michal-MK commented on dao/src/main/java/cz/idomatojde/dao/UserDao.java:
```diff
@@ -48,4 +49,11 @@
+    /**
+     * Adds offer as subscribed by user
+     * @param userId to be added to
+     * @param offer be added
```
`to be added to` and `be added` are not descriptive IMO

Michal-MK commented on rest/src/main/java/cz/idomatojde/rest/controllers/OfferController.java:
```diff
@@ -43,6 +45,27 @@ public OfferController(UserFacade userFacade, OfferFacade offers) {
+    }
+
+    @GetMapping("withSubscribedUser/{userId}")
+    ResponseEntity<List<OfferDTO>> withSubscribedUser(@PathVariable long userId) {
```
Not sure about the naming, probably `allSubscribedBy` would be better?

Michal-MK commented on rest/src/main/java/cz/idomatojde/rest/controllers/UserController.java:
```diff
@@ -50,6 +52,11 @@ public UserController(UserFacade users, OfferFacade offers) {
     }
 
+    @PostMapping(value = "signup")
+    ResponseEntity<Long> signup(@RequestBody RegisterUserDTO regDto) {
```
I would modify the base class and use the `regsiter` function there (another `final boolean allowUnauthenticatedRegister` perhaps? and set it only for the `UserController`)?

Michal-MK commented on ui/src/frontend/README.md:
```diff
@@ -0,0 +1,24 @@
+# frontend
```
I assume this is pre-generated? It does contain the `npm install` I mentioned earlier, could be merged into the main `README.md` in the root and removed or the main README should point to this one and the incorrect info should be deleted there.

Michal-MK commented on ui/src/frontend/src/components/HelloWorld.vue:
```diff
@@ -0,0 +1,22 @@
+<template>
```
File intended for testing?

Michal-MK committed:
Improved DataLoader
- more offers!
- better descriptions
- better names
- still bad code -.- ;)

Michal-MK committed:
Pagination
- added BaseDAO.findPaged and its implementation
- ^^ propagate this function up to the REST controllers
- implemented TODO on OfferController

Michal-MK committed:
Pagination test fix

Michal-MK committed:
Sample data improvements
- moved duplicate code into helper methods

Michal-MK committed:
Quick Format of UI and final data loading?
- removed the deprecated DTO delete call
- added try/catch around getById() and return null if no entity was found
- fixed pagination paths for REST controllers
- added getFiltered to Offer chain
- added more checks to SampleDataLoader to produce valid configurations
- signup to PUT from POST

JirkaVrbka committed:
Merge pull request #38 from JirkaVrbka/rest-for-ui

Rest changes for UI

JirkaVrbka committed:
Cleaning

JirkaVrbka committed:
Added js-cookie dependency

JirkaVrbka committed:
Storage module for api

JirkaVrbka committed:
removed deprecated script source

JirkaVrbka committed:
Exporting API calls to module

JirkaVrbka committed:
Correct regex check

JirkaVrbka committed:
Registration

JirkaVrbka committed:
updated requirements

JirkaVrbka committed:
Base of entries

Michal-MK committed:
Moved addSubscription to Offers from Users
- fixed subscriptions not saving for both ends
- improved JavaDoc

Michal-MK committed:
Changed the UI API call to addSubscription to match the new REST definition

JirkaVrbka committed:
select random cards for search

JirkaVrbka committed:
removed comments

JirkaVrbka committed:
Added api calls

JirkaVrbka committed:
Added local storage

JirkaVrbka committed:
extracted logic

JirkaVrbka committed:
component for subscriptions

JirkaVrbka committed:
Component for calendar

JirkaVrbka committed:
formatting and custom store

JirkaVrbka committed:
formatting

JirkaVrbka committed:
Extracted logic

JirkaVrbka committed:
Extracted logic + owner check for assignment

JirkaVrbka committed:
null check

JirkaVrbka committed:
Credit and admin check

JirkaVrbka committed:
Fixed doc

JirkaVrbka committed:
Removed todo

JirkaVrbka committed:
Added npm install

JirkaVrbka committed:
Autogenerating timetable
